tests/unit/test_questions/test_services.py

```diff
@@ -0,0 +1,164 @@
+        resolve_question(
+            post_child.question,
+            "yes",
+            actual_resolve_time=make_aware(datetime(2024, 11, 1)),
+        )
+        resolve_question(
+            post_parent.question,
+            "no",
+            actual_resolve_time=make_aware(datetime(2024, 11, 1)),
+        )
+        post_conditional.refresh_from_db()
+
+        assert post_conditional.resolved
+        assert post_conditional.actual_close_time == make_aware(datetime(2024, 11, 1))
+
+        assert post_conditional.conditional.question_no.resolution == "yes"
+        assert post_conditional.conditional.question_no.actual_close_time == make_aware(
+            datetime(2024, 11, 1)
+        )
+        assert (
+            post_conditional.conditional.question_no.actual_resolve_time
+            == make_aware(datetime(2024, 11, 1))
+        )
+
+        assert post_conditional.conditional.question_yes.resolution == "annulled"
+        assert (
+            post_conditional.conditional.question_yes.actual_close_time
+            == make_aware(datetime(2024, 11, 1))
+        )
+        assert (
+            post_conditional.conditional.question_yes.actual_resolve_time
+            == make_aware(datetime(2024, 11, 1))
+        )
+
+    def test_conditional_parent_resolved(
+        self, post_conditional, post_child, post_parent
+    ):
+        """
+        Both Condition and Child are resolved
+        """
+
+        assert not post_conditional.resolved
+
+        # Resolve condition
+        resolve_question(
+            post_parent.question,
+            "no",
+            actual_resolve_time=make_aware(datetime(2024, 11, 1)),
+        )
+        post_conditional.refresh_from_db()
+
+        assert not post_conditional.resolved
+        assert post_conditional.actual_close_time
+
+        assert not post_conditional.conditional.question_no.resolution
+        assert post_conditional.conditional.question_no.actual_close_time == make_aware(
+            datetime(2024, 11, 1)
+        )
+        assert not post_conditional.conditional.question_no.actual_resolve_time
+
+        assert post_conditional.conditional.question_yes.resolution == "annulled"
+        assert (
+            post_conditional.conditional.question_yes.actual_close_time
+            == make_aware(datetime(2024, 11, 1))
+        )
+        assert (
+            post_conditional.conditional.question_yes.actual_resolve_time
+            == make_aware(datetime(2024, 11, 1))
+        )
+
+    def test_conditional_child_resolved(self, post_conditional, post_child):
+        """
+        Both Condition and Child are resolved
+        """
+
+        assert not post_conditional.resolved
+
+        # Resolve condition
+        resolve_question(
+            post_child.question,
+            "no",
+            actual_resolve_time=make_aware(datetime(2024, 11, 1)),
+        )
+        post_conditional.refresh_from_db()
+
+        assert not post_conditional.resolved
```

Comment:
asserted already on line 140,
But I guess this does make sure resolving a child doesn't resolve the parent...